### PR TITLE
[ENG-2433] Include withdrawal requests in moderator digests

### PR DESCRIPTION
**Pull request by jwalz**
## Purpose
Put withdrawal notifications and submission notifications into the same moderator digest emails and eliminate the need for a separate template to produce the entry cards for withdrawals.

## Changes
* Fixed a bug in which we weren't storing digest emails for withdrawals
* Fixed a bug in which we were including `new_pending_withdrawal_request` events in the  generic user digest emails, thereby consuming them before they could be included in the moderator emails
* Explicitly passed the `new_pending_submissions` template when storing withdrawal requests instead of using the template inferred from the `new_pending_withdraw_requests` event.
* Updated the message when constructing withdraw messages to correctly note that withdrawal was requested instead of saying "submitted"
* Got rid of mocking in tests for `notify_moderator_registration_requests_withdrawal` that obscured the bug previously caused by the missing template for withdrawal emails.
* Wrote tests to confirm that submissions and withdrawals both appear in the moderator email digests and are not scooped up by the user digests.

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify that withdrawal requests show up alongside submissions in moderator emails
- Verify that withdrawal requests are clearly marked with "requested withdrawal"

What are the areas of risk?

Any concerns/considerations/questions that development raised?

## Ticket
https://openscience.atlassian.net/browse/ENG-2433

### CODE REVIEW

**Review by corbinSanders — APPROVED** [review]
LGTM!

File: website/reviews/listeners.py [hunk 1]
```diff
@@ -135,13 +135,15 @@ def reviews_withdraw_requests_notification_moderators(self, timestamp, context):
...
     # Set message
-    context['message'] = f'submitted "{resource.title}".'
+    context['message'] = f'has requested withdrawal of "{resource.title}".'
     # Set url for profile image of the submitter
     context['profile_image_url'] = get_profile_image_url(context['referrer'])
     # Set submission url
     context['reviews_submission_url'] = f'{DOMAIN}reviews/registries/{provider._id}/{resource._id}'
 
     email_transactional_ids = list(provider_subscription.email_transactional.all().values_list('guids___id', flat=True))
+#    print(email_transactional_ids)
```
Comment (brianjgeiger) [inline website/reviews/listeners.py:145]: Probably don't need this commented line anymore.

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
One stray comment.

File: website/reviews/listeners.py (hunk 1 above)
Comment (jwalz) [reply website/reviews/listeners.py]: I really need to ctrl+f for print statements before I put things in review 😂

**Comment by brianjgeiger** [comment]
Please update with develop so tests should pass.

**Review by brianjgeiger — APPROVED** [review]

**Pull request merged**